docs/dg/dev/backend-development/opentelemetry/overview.md

```diff
@@ -0,0 +1,416 @@
+    }
+
+}
+```
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getPathPatterns()` - allows to add a path pattern that shows where console command should look for you classes to cover them with hooks. OOTB all Spryker directories are covered together with `Pyz` namespace on the project. Be advised that you should not cover code that is autogenerated like Transfers.
+```php
+class OpentelemetryConfig extends AbstractBundleConfig
+{
+    public function getPathPatterns(): array
+    {
+        return [
+            '#^vendor/spryker/[^/]+/.*/.*/(Zed|Shared)/.*/(?!Persistence|Presentation)[^/]+/.*#',
+            '#^vendor/spryker/[^/]+/Glue.*#',
+            '#^vendor/spryker(?:/spryker)?-shop/[^/]+/.*#',
+            '#^vendor/spryker-eco/[^/]+/.*#',
+            '#^src/Pyz/.*#',
+        ];
+    }
+
+}
+```
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getOutputDir()` - specifies in what directory you want to put generated hooks. Default value is `src/Generated/OpenTelemetry/Hooks/`. The classmap file will also be added into this directory.
+```php
+class OpentelemetryConfig extends AbstractBundleConfig
+{
+    public function getOutputDir(): string
+    {
+        return APPLICATION_SOURCE_DIR . '/Generated/OpenTelemetry/Hooks/';
+    }
+
+}
+```
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::areOnlyPublicMethodsInstrumented()` - OOTB we cover with hooks only public methods of regular classes and all methods of Controller classes, but you can change that with this configuration method.
+```php
+class OpentelemetryConfig extends AbstractBundleConfig
+{
+    public function areOnlyPublicMethodsInstrumented(): bool
+    {
+        return true;
+    }
+
+}
+```
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getCriticalClassNamePatterns()` - some of the spans can be marked as `critical` in order to give them more priority on sampling. You can define what classes you are interested in for that. OOTB we have Controllers and Facades here. With the current implementation it's not a Regex, but just a string that we are looking for in the class name.
+```php
+class OpentelemetryConfig extends AbstractBundleConfig
+{
+    public function getCriticalClassNamePatterns(): array
+    {
+        return [
+            'Facade',
+            'Controller',
+        ];
+    }
+
+}
+```
+
+### Enable PHP extension
+
+Hooks processing requires you to have a few PHP extensions in place. Spryker has prepared a new PHP image, so you need to install nothing, but just enable it in your deploy file.
+
+```yaml
+namespace: spryker-otel
+tag: 'dev'
+
+environment: docker.dev
+image:
+    tag: spryker/php:8.3-alpine3.20-otel
+    php:
+        enabled-extensions:
+            - opentelemetry
+            - grpc
+            - protobuf
+```
+Be advised that `blackfire` extension can have a conflict with a `opentelemetry` one, so make sure that you are not using both in the same time. We also would not recommend to run it together with a `newrelic` one.
```

Comment:
Let's be more helpful here.
Blackfire can have conflicts - how is this visible if still both are enabled?
Is there an option how to add a switcher?